src/mqtt/mqtt_test.c

```diff
@@ -982,7 +982,7 @@ TEST( MqttTest, MQTT_Subscribe_Publish_With_Qos_1 )
             /* Timeout. */
             break;
         }
-        else if( ( receivedPubAck != 0 ) && ( strncmp( TEST_MQTT_TOPIC, incomingInfo.pTopicName, TEST_MQTT_TOPIC_LENGTH ) == 0 ) )
+        else if( ( receivedPubAck != 0 ) && ( incomingInfo.topicNameLength == TEST_MQTT_TOPIC_LENGTH ) )
```

Comment:
Why are we only comparing lengths? What happens if the topic name is not expected?

Reply:
We'll check the topic at below, like: 
`TEST_ASSERT_EQUAL_MEMORY( TEST_MQTT_TOPIC,
                              incomingInfo.pTopicName,
                              TEST_MQTT_TOPIC_LENGTH );`
                              
The check here is only to make sure that we receive at least one publish message from MQTT broker.